backend/check_stac_metadata/task.py

```diff
@@ -31,16 +28,7 @@ def main() -> int:
     validation_result_factory = ValidationResultFactory(hash_key)
     validator = STACDatasetValidator(s3_url_reader, validation_result_factory)
 
-    try:
-        validator.run(arguments.metadata_url)
-
-    except (AssertionError, ValidationError, ClientError) as error:
-        LOGGER.error(dumps({"success": False, "message": str(error)}))
-        return 1
-
-    validator.save(hash_key)
-
-    LOGGER.info(dumps({"success": True, "message": ""}))
+    validator.run(arguments.metadata_url, hash_key)
     return 0
```

Comment:
At this point we never actively return anything other than 0, so it's redundant. We can get rid of this line and simplify `sys.exit(main())` to just `main()`.